scripts/exglobal_extractvars.sh

```diff
@@ -0,0 +1,69 @@
+#! /usr/bin/env bash                                                                                                                                                                    
+################################################################################
+#   Script:    
+#
+source "${USHgfs}/preamble.sh"
+
+# Programs used
+export WGRIB2=${WGRIB2:-${wgrib2_ROOT}/bin/wgrib2}
+
+# Scripts used
+EXTRCTVARA=${EXTRCTVARA:-"${USHgfs}/gefs_atmos_extractvars.sh"}
+EXTRCTVARO=${EXTRCTVARO:-"${USHgfs}/gefs_ocnice_extractvars.sh"}
+EXTRCTVARW=${EXTRCTVARW:-"${USHgfs}/gefs_wav_extractvars.sh"}
+
+#Paramater Tables used
+export varlist_2d=${varlist_2d:-"${PARMgfs}/product/gefs_shortparmlist_2d.parm"} #Parameter table for surface variables
+export varlist_3d=${varlist_3d:-"${PARMgfs}/product/gefs_shortparmlist_3d_h.parm"} #Parameter table for upper air instantaneous variables
+export varlist_3d_d=${varlist_3d_d:-"${PARMgfs}/product/gefs_shortparmlist_3d_d.parm"} #Parameter table for upper air daily-averaged variables
+export varlist_wav=${varlist_wav:-"${PARMgfs}/product/gefs_wav_shortparmlist.parm"} #Parameter table for wave variables
+export varlist_ocn_netcdf=${varlist_ocn_netcdf:-"${PARMgfs}/product/gefs_ocn_shortparmlist.parm"} #Parameter table for wave variables
+export varlist_ice_netcdf=${varlist_ice_netcdf:-"${PARMgfs}/product/gefs_ice_shortparmlist.parm"} #Parameter table for wave variables
+export varlist_ocn_grib2=${varlist_ocn_grib2:-"${PARMgfs}/product/gefs_ocn_shortparmlist_grib2.parm"} #Parameter table for wave variables
+export varlist_ice_grib2=${varlist_ice_grib2:-"${PARMgfs}/product/gefs_ice_shortparmlist_grib2.parm"} #Parameter table for wave variables  
+
+#Extract variables for atmosphere
+if [[ "${DO_ATM}" == "YES" ]];then
+  if [[ ! -d "${DATA}/mem${ENSMEM}_atmos" ]]; then 
+    mkdir -p "${DATA}/mem${ENSMEM}_atmos" 
+  fi
+  ${EXTRCTVARA} "${DATA}/mem${ENSMEM}_atmos"
+  cp -pr "${DATA}/mem${ENSMEM}_atmos/f2d" "${COM_RFCST_PROD_ATMOS}"
+  cp -pr "${DATA}/mem${ENSMEM}_atmos/f3d" "${COM_RFCST_PROD_ATMOS}"
+fi
+
+#Extract variables for ocean
+if [[ "${DO_OCN}" == "YES" ]];then
+  export component_name="ocn"
+  if [[ ! -d "${DATA}/mem${ENSMEM}_ocn" ]]; then 
+    mkdir -p "${DATA}/mem${ENSMEM}_ocn" 
+  fi
+  if [[ "${ocn_dataformat}" == "netcdf" ]]; then varlist_ocn=${varlist_ocn_netcdf}; fi
+  if [[ "${ocn_dataformat}" == "grib2" ]]; then varlist_ocn=${varlist_ocn_grib2}; fi  
```

Comment:
Perhaps a test here for an invalid/undefined `ocn_dataformat` would be a good idea:

```suggestion
  if [[ "${ocn_dataformat}" == "netcdf" ]]; then
    varlist_ocn=${varlist_ocn_netcdf}
  elif [[ "${ocn_dataformat}" == "grib2" ]]; then
    varlist_ocn=${varlist_ocn_grib2}
  else
    echo "FATAL ERROR: Invalid ocean data format provided (${ocn_dataformat})"
    export err=1; ${errchk}
  fi
```

Reply:
@DavidHuber-NOAA Where is `errchk` defined? Is it ok to define it in `JGLOBAL_EXTRACTVARS`, such as `export errchk=${errchk:-err_chk}`?

Reply:
@EricSinsky-NOAA  Apologies, I copied this from another script.  Yes, defining it in the J-Job would be fine.  I would also be OK with using `err_chk` here:

```bash
    export err=1; err_chk
```

Reply:
Thank you, @DavidHuber-NOAA. I think I'll just use `err_chk` here.

Reply:
@DavidHuber-NOAA When `err_chk` is executed, it seems to function mostly as expected, however, I get the following message: `cat: OUTPUT.185562: No such file or directory`
This is not fatal, but is there a fix for this?

Reply:
@EricSinsky-NOAA I am not sure what the origin of this message is.  I'm guessing it is coming from `err_exit`, but I'm not sure.  Could you send me the location of a log file?  I'll take a look.

Reply:
@DavidHuber-NOAA You can find my log file here on dogwood: /lfs/h2/emc/ens/noscrub/eric.sinsky/errlog/extractvars_mem000.log

Reply:
I think this can be ignored.  I suspect `pgmout` is something that is only generated during production.  Is that correct @aerorahul?